test/unit/deferred.js

```diff
@@ -759,10 +756,8 @@ QUnit.test( "jQuery.when - joined", function( assert ) {
 				expected = shouldResolve ? [ 1, 1 ] : [ 0, undefined ],
 				expectedNotify = shouldNotify && [ willNotify[ id1 ], willNotify[ id2 ] ],
 				code = "jQuery.when( " + id1 + ", " + id2 + " )",
-				context1 = defer1 && jQuery.isFunction( defer1.promise ) ? defer1.promise() :
-					( defer1.then ? window : undefined ),
-				context2 = defer2 && jQuery.isFunction( defer2.promise ) ? defer2.promise() :
-					( defer2.then ? window : undefined );
+				context1 = defer1 && jQuery.isFunction( defer1.promise ) ? defer1.promise() : window,
+				context2 = defer2 && jQuery.isFunction( defer2.promise ) ? defer2.promise() : window;
```

Comment:
Not since at least jQuery 1.8.3: http://jsfiddle.net/av4shtg8/ . I complained about the `cast`/`all` mixing in #2018, but lost.

This test code should not be altered, and the behavior change should be reverted. If anything, the problem is that the [Callbacks `apply`](https://github.com/timmywil/jquery/blob/when-2546/src/callbacks.js#L80) converts our `undefined` context into the global object, but there's nothing we can do about that language-level operation.